api/v1beta2/ibmvpccluster_types.go

```diff
@@ -98,26 +110,108 @@ type AdditionalListenerSpec struct {
+	// +required
+	HealthTimeout int64 `json:"healthTimeout"`
+
+	// healthType defines the protocol used for health checks.
+	// +required
+	HealthType string `json:"healthType"`
+
+	// healthMonitorURL defines the URL to use for health monitoring.
+	// +optional
+	HealthMonitorURL *string `json:"healthMonitorURL,omitempty"`
+
+	// healthMonitorPort defines the port to perform health monitoring on.
+	// +optional
+	HealthMonitorPort *int64 `json:"healthMonitorPort,omitempty"`
+}
+
+// ImageSpec defines the desired state of the VPC Custom Image resources for the cluster.
+// +kubebuilder:validation:XValidation:rule="(!has(self.cosInstance) && !has(self.cosBucket) && !has(self.cosObject)) || (has(self.cosInstance) && has(self.cosBucket) && has(self.cosObject))",message="if any of cosInstance, cosBucket, or cosObject are specified, all must be specified"
+// +kubebuilder:validation:XValidation:rule="has(self.name) || has(self.crn) || (has(self.cosInstance) && has(self.cosBucket) && has(self.cosObject))",message="an existing image name or crn must be provided, or to create a new image the cos resources must be provided, with or without a name"
+type ImageSpec struct {
+	// name is the name of the desired VPC Custom Image.
+	// +optional
+	Name *string `json:"name,omitempty"`
+
+	// crn is the IBM Cloud CRN of the existing VPC Custom Image
+	// +optional
+	CRN *string `json:"crn,omitempty"`
+
+	// cosInstance is the name of the IBM Cloud COS Instance containing the source of the image, if necessary.
+	// +optional
+	COSInstance *string `json:"cosInstance,omitempty"`
+
+	// cosBucket is the name of the IBM Cloud COS Bucket containing the source of the image, if necessary.
+	// +optional
+	COSBucket *string `json:"cosBucket,omitempty"`
+
+	// cosBucketRegion is the COS region the bucket is in.
+	// +optional
+	COSBucketRegion *string `json:"cosBucketRegion,omitempty"`
 
+	// cosObject is the name of a IBM Cloud COS Object used as the source of the image, if necessary.
+	// +optional
+	COSObject *string `json:"cosObject,omitempty"`
+
+	// operatingSystem is the Custom Image's Operating System name.
+	// +optional
+	OperatingSystem *string `json:"operatingSystem,omitempty"`
+
+	// resourceGroup is the Resource Group to create the Custom Image in.
+	// +optional
+	ResourceGroup *IBMCloudResourceReference `json:"resourceGroup,omitempty"`
+}
+
+// VPCNetworkSpec defines the desired state of the network resources for the cluster for extended VPC Infrastructure support.
+type VPCNetworkSpec struct {
 	// controlPlaneSubnets is a set of Subnet's which define the Control Plane subnets.
 	// +optional
 	ControlPlaneSubnets []Subnet `json:"controlPlaneSubnets,omitempty"`
 
-	// resourceGroup is the name of the Resource Group containing all of the newtork resources.
+	// loadBalancers is a set of VPC Load Balancer definitions to use for the cluster.
+	// +optional
+	LoadBalancers []VPCLoadBalancerSpec `json:"loadbalancers,omitempty"`
```

Comment:
Since we are defining here as well, We may need to add proper comment for VPCSpec.ControlplaneLoadbalancer.

Reply:
Sure, I can try to add something to the old field.

Reply:
Added some details about legacy use versus extended support.